guacamole-docker/bin/start.sh

```diff
@@ -141,16 +141,39 @@ environment variables:
 
     MYSQL_DATABASE     The name of the MySQL database to use for Guacamole
                        authentication.
-END
+END`
```

Comment:
I believe that trailing backtick here will result in the `END` failing to match the `<<END`, as the shell will be searching for a line that contains nothing but `END`. Can you confirm whether this works as expected?

Reply:
My very rough example compose file for a guacamole stack:

```yaml
version: "3.1"
services:
  daemon:
    image: guacamole/guacd:latest
    networks:
      - guac

  client:
    image: wojoinc/guac-test:latest
    networks:
      - guac
      - db
    ports:
      - "8081:8080"

    environment:
      GUACD_HOSTNAME: "daemon"
      GUACD_PORT: "4822"
      MYSQL_HOSTNAME: "db"
      #MYSQL_DATABASE_FILE: "/run/secrets/guac-mysql-db"
      MYSQL_PORT: "3306"
      MYSQL_USER_FILE: "/run/secrets/guac-mysql-user"
      MYSQL_PASSWORD_FILE: "/run/secrets/guac-mysql-password"

    secrets:
      - guac-mysql-db
      - guac-mysql-user
      - guac-mysql-password

  db:
    image: mariadb:latest
    networks:
      - db
    environment:
      MYSQL_ROOT_PASSWORD_FILE: "/run/secrets/guac-mysql-root-password"
      MYSQL_DATABASE: "guacamole"
      MYSQL_USER_FILE: "/run/secrets/guac-mysql-user"
      MYSQL_PASSWORD: "/run/secrets/guac-mysql-password"
      MYSQL_ROOT_HOST: "localhost"
    volumes:
      - "db_data:/var/lib/mysql"
      - "./initdb.sql:/docker-entrypoint-initdb.d/dump.sql"
    secrets:
      - guac-mysql-root-password
      - guac-mysql-db
      - guac-mysql-user
      - guac-mysql-password

networks:
  guac:

  db:

secrets:
  guac-mysql-root-password:
    external: true
  guac-mysql-db:
    external: true
  guac-mysql-user:
    external: true
  guac-mysql-password:
    external: true

volumes:
  db_data:
```

When omitting MYSQL_DATABASE or MYSQL_DATABASE_FILE, this causes the container to fail to spin up, and the logs present the error message from before calling associate_mysql() located at the bottom of the file,
```
FATAL: No authentication configured
-------------------------------------------------------------------------------
The Guacamole Docker container needs at least one authentication mechanism in
order to function, such as a MySQL database, PostgreSQL database, or LDAP
directory.  Please specify at least the MYSQL_DATABASE or POSTGRES_DATABASE
environment variables, or check Guacamole's Docker documentation regarding
configuring LDAP and/or custom extensions.
```

If you omit MYSQL_USER for example however, it does appear to read beyond the trailing backtick. I made the change to remove it, and I am rebuilding a test image now to see if it still works with that back tick removed. Thanks for letting me know!